x-pack/plugins/observability/public/pages/cases/use_fetch_alert_detail.test.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks';
+import { kibanaStartMock } from '../../utils/kibana_react.mock';
+import { TopAlert } from '../alerts';
+import * as pluginContext from '../../hooks/use_plugin_context';
+import { createObservabilityRuleTypeRegistryMock } from '../..';
+import { PluginContextValue } from '../../context/plugin_context';
+import { useFetchAlertDetail } from './use_fetch_alert_detail';
+
+const mockUseKibanaReturnValue = kibanaStartMock.startContract();
+
+jest.mock('../../utils/kibana_react', () => ({
+  __esModule: true,
+  useKibana: jest.fn(() => mockUseKibanaReturnValue),
+}));
+
+describe('useFetchAlertDetail', () => {
+  const getResult = {
+    'kibana.alert.rule.category': 'Metric threshold',
+    'kibana.alert.rule.consumer': 'infrastructure',
+    'kibana.alert.rule.execution.uuid': 'e62c418d-734d-47e7-bbeb-e6f182f5fb45',
+    'kibana.alert.rule.name': 'A super rule',
+    'kibana.alert.rule.producer': 'infrastructure',
+    'kibana.alert.rule.rule_type_id': 'metrics.alert.threshold',
+    'kibana.alert.rule.uuid': '69411af0-82a2-11ec-8139-c1568734434e',
+    'kibana.space_ids': ['default'],
+    'kibana.alert.rule.tags': [],
+    '@timestamp': '2022-01-31T18:20:57.204Z',
+    'kibana.alert.reason': 'Document count has reported no data over the past 1h for *',
```

Comment:
Nit: We recently updated the reason message as part of https://github.com/elastic/kibana/issues/121272. Here's the new format `Document count reported no data in the last 1 hour for all hosts`.